### PR TITLE
Various CI fixes

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Various suggestions on CI fixes for `ocaml-grpc`:
* b1bd2004ba77f46c96266f6bb86486cbd2db4219 adds the latest ocaml-format version to the config file
* f4fe0a48a8096a379d05a55b0bdbf9016741b4ff creates an opam package for examples and benchmarking dependencies. I also removed the opam.template files since they are empty and were not doing anything.
* c752475db4e7c495b08a73b22d0fde0c86e99359 updates the dune version and attaches the examples to the correct package.

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(lang dune 2.7)
+(lang dune 3.6)
```
Comment (quernd) [inline dune-project:1]: Just wondering about the bump to 3.6 here, is there some specific functionality that isn't available in 3.0?

**Review by quernd — APPROVED** [review]
Thank you @tmcgilchrist, much appreciated! Great to see the green checkmark again. Just wondering if dune 3.6 is necessary or if we can stay on 3.0?

File: dune-project (hunk 1 above)
Comment (tmcgilchrist) [reply dune-project:1]: There are 3 main reasons:
 * newer Dune versions have important Windows fixes (I'm planning to validate Windows support when I get around to it)
 * dune 3.2 provides some fixes for mdx, which I was planning to use for writing/validating documentation
 * dune 3.6/3.7 still support OCaml 4.08, a reasonable lower bound for OCaml versions
 
Using `(lang dune 3.0)` and installing dune from opam will give you the newer version of dune. But there would potentially be bugs if you used the lower bounds for all packages.

Comment (tmcgilchrist) [reply dune-project:1]: Personally I'm in the habit of using the latest dune versions where possible, I know not everyone has that option. If there is a need to stick to dune 3.0 then I can fix things like https://github.com/tmcgilchrist/ocaml-grpc/pull/1/commits/ac1b56fb40f66ec1d932a297b1b022d51f6b8dde.

Comment (quernd) [reply dune-project:1]: Thanks, you convinced me 👍

**Pull request merged**